railties/test/generators/app_generator_test.rb

```diff
@@ -56,6 +56,13 @@ class AppGeneratorTest < Rails::Generators::TestCase
   # brings setup, teardown, and some tests
   include SharedGeneratorTests
 
+  def assert_generates_with_bundler(options = {})
+    generator([destination_root], options)
+    assert_called_with(generator, :bundle_command, [ ['install'], ["exec spring binstub --all"] ]) do
+      quietly { generator.invoke_all }
+    end
```

Comment:
Eh, I don't know if we should call that private method. How about this?

```ruby
def assert_generates_with_bundler(options = {})
  generator([destination_root], options)
  
  command_check = -> command do
    @install_called ||= 0

    case command
    when 'install'
      @install_called += 1
      assert_equal 1, @install_called, "install expected to be called once, but was called #{@install_called} times"
    when 'exec spring binstub --all'
      # Called when running tests with spring, let through unscathed.
    end
  end

  generator.stub :bundle_command, command_check do
    quietly { generator.invoke_all }
  end
end
```

It might need to be tweaked but that's a possible direction :grin: